vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ContentBlockManagerTest.java

```diff
@@ -82,8 +101,65 @@ public void test_deleteTextBlockById_forNonExistentId() throws BlockDoesNotExist
         Mockito.doThrow(EmptyResultDataAccessException.class).when(textBlockRepo).deleteById(textBlockId);
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(textBlockRepo).existsById(textBlockId);
         managerToTest.deleteTextBlockById(textBlockId, "slideId_1");
     }
+    
+    @Test(expected = BlockDoesNotExistException.class)
+    public void test_deleteSpaceBlockById_forNonExistentId() throws BlockDoesNotExistException {
+        String spaceBlockId = "notARealId";
+        when(contentBlockRepository.findById("notARealId")).thenReturn(Optional.empty());
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(spaceBlockRepo).deleteById(spaceBlockId);
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(textBlockRepo).existsById(spaceBlockId);
+        managerToTest.deleteSpaceBlockById(spaceBlockId,"slideId_1");
+    }
+    
+    @Test
+    public void test_deleteSpaceBlock_valid() throws BlockDoesNotExistException {
+        String spaceBlockId = "realId";
+        when(contentBlockRepository.findById(spaceBlockId)).thenReturn(Optional.of(contentBlock));
+        managerToTest.deleteSpaceBlockById(spaceBlockId,"slideId_1");
+        Mockito.verify(spaceBlockRepo).deleteById(spaceBlockId);
+        
+    }
+    
+    @Test
+    public void test_addSpaceBlock_success() {
+        String slideId = "slideId";
+        ISlide slide = null;
+        String title = "this is a space block";
+        Integer contentOrder = 2;
+        Space space = new Space();
+        ISpaceBlock spaceBlock = new SpaceBlock();
+        Mockito.when(slideManager.getSlide(slideId)).thenReturn(slide);
+        Mockito.when(spaceBlockRepo.save((SpaceBlock)spaceBlock)).thenReturn((SpaceBlock) spaceBlock);
+        Mockito.when(spaceBlockFactory.createSpaceBlock(slide, title, (ISpace)space)).thenReturn(spaceBlock);
+        ISpaceBlock createdBlock = managerToTest.createSpaceBlock(slideId, title, contentOrder, space);
+        Assert.assertEquals(createdBlock.getContentOrder(), contentOrder);
+    }
+    
+    @Test
+    public void test_getSpaceBlock_success() {
+        String spaceId="spaceId";
+        SpaceBlock spaceBlock = new SpaceBlock();
+        spaceBlock.setId(spaceId);
+        when(spaceBlockRepo.findById(spaceId)).thenReturn(Optional.of(spaceBlock));
+        ISpaceBlock retrievedSpaceBlock = managerToTest.getSpaceBlock(spaceId);
+        assertEquals(spaceId, retrievedSpaceBlock.getId());
+    }
+    
+    @Test(expected = Exception.class)
+    public void test_getSpaceBlock_forNonExistentId() throws BlockDoesNotExistException {
+    	
+    	String spaceId = "notRealId";
+    	SpaceBlock spaceBlock = new SpaceBlock();
+        spaceBlock.setId(spaceId);
+        Mockito.doThrow(EmptyResultDataAccessException.class).when(spaceBlockRepo).findById(spaceId);
+        ISpaceBlock retrievedSpaceBlock = managerToTest.getSpaceBlock(spaceId);
+        assertNull(retrievedSpaceBlock);
+        
+      
+    }
```

Comment:
please reread our test guidelines. Test methods don't follow our naming convention.

Reply:
@jdamerow are you refering to this method - test_getSpaceBlock_forNonExistentId() ? If yes I will change it to test_getSpaceBlock_missingId()